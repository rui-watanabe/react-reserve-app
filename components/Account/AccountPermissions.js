import React from 'react';
import axios from 'axios';
import baseUrl from '../../utils/baseUrl';
import cookie from 'js-cookie';

function AccountPermissions() {
  const [users, setUsers] = React.useState([])

  React.useEffect(() => {
    getUsers();
  },[]);

  async function getUsers(){
    const url = `${baseUrl}/api/users`;
    const token = cookie.get('token');
    const payload = { headers: { Authorization: token } };
    const response = await axios.get(url, payload);
    console.log(response.data)
    // setUsers(response.data);
  }

  return(
    <>
    </>
  )
}

export default AccountPermissions;
