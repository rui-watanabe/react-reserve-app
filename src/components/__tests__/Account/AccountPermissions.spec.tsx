import React from 'react';
import renderer from 'react-test-renderer';
import AccountPermissions from '../../Account/AccountPermissions';

test('AccountPermissions', () => {
  const component = renderer.create(<AccountPermissions />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
