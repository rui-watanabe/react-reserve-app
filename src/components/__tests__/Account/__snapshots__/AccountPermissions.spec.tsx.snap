// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountPermissions 1`] = `
<div
  style={
    Object {
      "margin": "2em 0",
    }
  }
>
  <h2
    className="ui header"
  >
    <i
      aria-hidden="true"
      className="settings icon"
      onClick={[Function]}
    />
    User Permissions
  </h2>
  <table
    className="ui celled definition compact table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        />
        <th
          className=""
        >
          Name
        </th>
        <th
          className=""
        >
          Email
        </th>
        <th
          className=""
        >
          Joined
        </th>
        <th
          className=""
        >
          Updated
        </th>
        <th
          className=""
        >
          Role
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    />
  </table>
</div>
`;
